src/main/java/de/adorsys/keycloak/config/repository/ClientRepository.java

```diff
@@ -182,12 +182,24 @@ public void createAuthorizationResource(String realmName, String id, ResourceRep
 
     public void updateAuthorizationResource(String realmName, String id, ResourceRepresentation resource) {
         ClientResource clientResource = getResourceById(realmName, id);
-        clientResource.authorization().resources().resource(resource.getId()).update(resource);
+        String resourceId = getResourceId(clientResource, resource.getName());
+        clientResource.authorization().resources().resource(resourceId).update(resource);
     }
 
-    public void removeAuthorizationResource(String realmName, String id, String resourceId) {
+    public void removeAuthorizationResource(String realmName, String id, String resourceName) {
         ClientResource clientResource = getResourceById(realmName, id);
-        clientResource.authorization().resources().resource(resourceId).remove();
+        String resourceId = getResourceId(clientResource, resourceName);
+        if (resourceId != null) {
```

Comment:
No warning, error or log when the resourceId was not found?

Reply:
In my opinion idempotency requires not existing entries not to throw errors when they get deleted. I am not sure if this needs a debug log or not but it doesn't need a warn log.

Reply:
Hi @bohmber,

you are free to contribute as well. 😉